### PR TITLE
introduce ParquetCacheableFileReader support cache BinaryDataFiber

### PR DESCRIPTION
**Pull request by ivoson**
## What changes were proposed in this pull request?
Introduce ParquetCacheableFileReader which supports cache BinaryDataFiber. This is a subtask of binary cache support as discusse in  [#1047 ](https://github.com/Intel-bigdata/OAP/pull/1047).

## How was this patch tested?
add new ut.

### CODE REVIEW

**Comment by ivoson** [comment]
cc @LuciferYang

File: src/main/java/org/apache/parquet/hadoop/MrOapRecordReader.java [hunk 1]
```diff
@@ -61,10 +61,10 @@ public T getCurrentValue() throws IOException, InterruptedException {
 
     @Override
     public void initialize() throws IOException, InterruptedException {
-      ParquetFileReader parquetFileReader = null;
+      ParquetCacheableFileReader parquetFileReader = null;
       try {
         parquetFileReader =
-                ParquetFileReader.open(configuration, file, footer.toParquetMetadata());
+                new ParquetCacheableFileReader(configuration, file, footer.toParquetMetadata());
```
Comment (LuciferYang) [inline src/main/java/org/apache/parquet/hadoop/MrOapRecordReader.java]: indent

File: src/main/java/org/apache/parquet/hadoop/ParquetCacheableFileReader.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    super(conf, file, footer);
+    this.binaryCacheEnabled =
+      conf.getBoolean(OapConf$.MODULE$.OAP_PARQUET_BINARY_DATA_CACHE_ENABLED().key(), false);
+
+    if (binaryCacheEnabled) {
+      this.dataFile = new ParquetDataFile(file.toUri().toString(), new StructType(), conf);
+      List<String[]> paths = footer.getFileMetaData().getSchema().getPaths();
+      int size = paths.size();
+      pathToColumnIndexMap = Maps.newHashMapWithExpectedSize(size);
+      for (int i = 0; i < size; i++) {
```
Comment (LuciferYang) [inline src/main/java/org/apache/parquet/hadoop/ParquetCacheableFileReader.java]: Maybe we should add some TODO: The pathToColumnIndexMap contains all `columnName -> columnId` for every file,  but we just need `required columns`,。

**Comment by LuciferYang** [comment]
related to #1124

File: src/main/java/org/apache/parquet/hadoop/ParquetCacheableFileReader.java (hunk 2 above)
Comment (ivoson) [reply src/main/java/org/apache/parquet/hadoop/ParquetCacheableFileReader.java]: ok,  i will add a toto here first, and then handle it later.

**Comment by ivoson** [comment]
@LuciferYang thanks for the review, the last commit address the comments you left

**Comment by zhixingheyi-tian** [comment]
@ivoson @LuciferYang 

Sorry for delay reponse for #1118 

https://github.com/Intel-bigdata/OAP/pull/1118/files#r364077887

Thanks.

**Pull request merged**